pkg/virt-launcher/virtwrap/converter/archconverter.go

```diff
@@ -0,0 +1,49 @@
+/* Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright the KubeVirt Authors.
+ *
+ */
+
+package converter
+
+import (
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+)
+
+type ArchConverter interface {
+	AddGraphicsDevice(vmi *v1.VirtualMachineInstance, domain *api.Domain, c *ConverterContext)
+	SCSIController(c *ConverterContext, driver *api.ControllerDriver) api.Controller
+	isUSBNeeded(c *ConverterContext, vmi *v1.VirtualMachineInstance) bool
+}
+
+func GetArchConverter(arch string) ArchConverter {
+	switch {
+	case isARM64(arch):
+		return archConverterARM64{}
+	case isPPC64(arch):
+		return archConverterPPC64{}
+	default:
```

Comment:
Please add the validation of the arch string.

Reply:
I would rather not do that for 2 reasons:
1. Adding the error makes the parts of the code where we call the functions more complicated, as i can't just slot a call to ArchConverter in as a replacement. Additionally especially `isUSBNeeded` does not throw an error itself. Though this is a minor reason
2. My main reason for not throwing an error is that we do not do that anywhere else in the code. In all cases the default behaviour is always to assume that the arch is x86_64. I don't see a reason to change it here, as it was also here the case

Reply:
I also think that validation would be valuable here.
@jschintag I would suggest something like:
```go
	switch {
	case isARM64(arch):
		return archConverterARM64{}
	case isPPC64(arch):
		return archConverterPPC64{}
	case isAMD64(arch):
		return archConverterAMD64{}
	default:
		log.Warningf("Trying to create an arch converter from an unknown arch: %s. falling back to AMD64", arch)
		return archConverterAMD64{}
```

This way you will able to do things like `ArchConverter(arch).isUSBNeeded()`, and the default would indeed be AMD64. But, on the other hand, an unknown arch being passed would still be somewhat visible.

My main concern is that without this a bug would probably be very hard to solve.

WDYT?